controller/src/main/java/io/pravega/controller/store/stream/PravegaTablesStreamMetadataStore.java

```diff
@@ -73,19 +76,25 @@
         this.completedTxnGC = new ZKGarbageCollector(COMPLETED_TXN_GC_NAME, zkStoreHelper, this::gcCompletedTxn, gcPeriod);
         this.completedTxnGC.startAsync();
         this.completedTxnGC.awaitRunning();
+        this.completedTxnGCRef = new AtomicReference<>(completedTxnGC);
         this.counter = new ZkInt96Counter(zkStoreHelper);
         this.storeHelper = new PravegaTablesStoreHelper(segmentHelper, authHelper, executor);
         this.executor = executor;
     }
 
-    private CompletableFuture<Void> gcCompletedTxn() {
+    @VisibleForTesting 
+    CompletableFuture<Void> gcCompletedTxn() {
         List<String> batches = new ArrayList<>();
         return withCompletion(storeHelper.expectingDataNotFound(storeHelper.getAllKeys(COMPLETED_TRANSACTIONS_BATCHES_TABLE)
                                                          .collectRemaining(batches::add)
                                                          .thenApply(v -> {
                                                                      // retain latest two and delete remainder.
                                                                      if (batches.size() > 2) {
-                                                                         return batches.subList(0, batches.size() - 2);
+                                                                         List<String> list = batches
+                                                                                 .stream().sorted(Comparator.comparingLong(Long::parseLong))
+                                                                                 .collect(Collectors.toList());
+
+                                                                         return list.subList(0, batches.size() - 2);
```

Comment:
What's the expected behavior in the case that `batches.size() < 2`? This is going to throw `IndexOutOfBounds`, no?

Reply:
This is the code path if batch.size > 2